test/run_test.py

```diff
@@ -1045,11 +1045,12 @@ def handle_log_file(
                 print_to_stderr(line.strip())
         print_to_stderr("")
         return
-    # otherwise: print entire file and then remove it
-    print_to_stderr(f"\nPRINTING LOG FILE of {test} ({file_path})")
-    print_to_stderr(full_text)
-    print_to_stderr(f"FINISHED PRINTING LOG FILE of {test} ({file_path})\n")
-    os.remove(file_path)
+
+    # otherwise: print entire file
+    print_to_stderr(f"\nPRINTING LOG FILE of {test} ({new_file})")
+    for line in full_text.splitlines():
```

Comment:
This's probably ok.  On the other hand, I think if we could open in text mode, i.e. `with open(file_path, "r") as f` then the `\r\n` newlines on Windows will be automatically handled I think.  I vaguely remember that opening in binary mode then decoding with ignore error is to skip over bad characters.  Then there is also `errors=ignore` when opening a file, I wonder why we don't use that.